tests/xharness/Harness.cs

```diff
@@ -219,6 +221,7 @@ public Harness (IResultParser resultParser, HarnessAction action, HarnessConfigu
 				SdkRoot = config ["XCODE_DEVELOPER_ROOT"] ?? configuration.SdkRoot;
 
 			processManager = new ProcessManager (XcodeRoot, MlaunchPath);
+			TunnelBore = new TunnelBore (processManager);
```

Comment:
what's the cost of creating this if `UseTcpTunnel` turns out to be `false` ?

Reply:
It is an empty dictionary. But we can simply not create it, and if it is null it wont be executed.